tests/filecheck/transforms/distribute-stencil.mlir

```diff
@@ -26,7 +26,7 @@ builtin.module {
 
 // CHECK:         func.func @offsets(%0 : !stencil.field<[-4,68]x[-4,68]x[-4,68]xf64>, %1 : !stencil.field<[-4,68]x[-4,68]x[-4,68]xf64>, %2 : !stencil.field<[-4,68]x[-4,68]x[-4,68]xf64>) {
 // CHECK-NEXT:      %3 = stencil.load %0 : !stencil.field<[-4,68]x[-4,68]x[-4,68]xf64> -> !stencil.temp<[-1,33]x[-1,33]x[0,32]xf64>
-// CHECK-NEXT:      "dmp.swap"(%3) {"topo" = #dmp.topo<2x2x2>, "swaps" = [#dmp.exchange<at [32, 0, 0] size [1, 32, 32] source offset [-1, 0, 0] to [1, 0, 0]>, #dmp.exchange<at [-1, 0, 0] size [1, 32, 32] source offset [1, 0, 0] to [-1, 0, 0]>, #dmp.exchange<at [0, 32, 0] size [32, 1, 32] source offset [0, -1, 0] to [0, 1, 0]>, #dmp.exchange<at [0, -1, 0] size [32, 1, 32] source offset [0, 1, 0] to [0, -1, 0]>]} : (!stencil.temp<[-1,33]x[-1,33]x[0,32]xf64>) -> ()
+// CHECK-NEXT:      "dmp.swap"(%3) {"topo" = #dmp.topo<2x2x2>, "strat" = #dmp.grid_slice_2d<[#builtin.int<2>, #builtin.int<2>, #builtin.int<2>], false>, "swaps" = [#dmp.exchange<at [32, 0, 0] size [1, 32, 32] source offset [-1, 0, 0] to [1, 0, 0]>, #dmp.exchange<at [-1, 0, 0] size [1, 32, 32] source offset [1, 0, 0] to [-1, 0, 0]>, #dmp.exchange<at [0, 32, 0] size [32, 1, 32] source offset [0, -1, 0] to [0, 1, 0]>, #dmp.exchange<at [0, -1, 0] size [32, 1, 32] source offset [0, 1, 0] to [0, -1, 0]>]} : (!stencil.temp<[-1,33]x[-1,33]x[0,32]xf64>) -> ()
```

Comment:
Is there duplicate information between strat and topo?

Reply:
@AntonLydike Do you have an opinion on this?
My understanding is that they do not represent the same thing (rank topology and gride slicing respectively) but that those happen to coincide in the current strategies.
I'd be fine just making the topo part of the strategy or keep as-is: do you think it's worth keeping them decoupled?

Reply:
Okay, went with making topo a part of strategy anyway.
As pointed by @n-io, returning a topology is part of a strategy's interface anyway!